File: app/src/main/java/com/booboot/vndbandroid/util/view/FrameLayoutKeyboardSupport.java
package com.booboot.vndbandroid.util.view;

import android.content.Context;
import android.util.AttributeSet;
import android.view.WindowInsets;
import android.widget.FrameLayout;

public class FrameLayoutKeyboardSupport extends FrameLayout {
    public FrameLayoutKeyboardSupport(Context context) {
        super(context);
    }

    public FrameLayoutKeyboardSupport(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public FrameLayoutKeyboardSupport(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public FrameLayoutKeyboardSupport(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
    }

    @Override
    public WindowInsets onApplyWindowInsets(WindowInsets insets) {
        /* removes the paddings induced by fitsSystemWindows="true" on all children views */
        setPadding(0, 0, 0, insets.getSystemWindowInsetBottom());
        return insets.consumeSystemWindowInsets();
    }
}

File: app/src/main/java/com/booboot/vndbandroid/util/view/FrameLayoutKeyboardSupport.kt
package com.booboot.vndbandroid.util.view

import android.content.Context
import android.util.AttributeSet
import android.view.WindowInsets
import android.widget.FrameLayout

class FrameLayoutKeyboardSupport : FrameLayout {
    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int, defStyleRes: Int) : super(context, attrs, defStyleAttr, defStyleRes)

    override fun onApplyWindowInsets(insets: WindowInsets): WindowInsets {
        /* removes the paddings induced by fitsSystemWindows="true" on all children views */
        setPadding(0, 0, 0, insets.systemWindowInsetBottom)
        return insets.consumeSystemWindowInsets()
    }
}